sktime/classification/model_evaluation/_functions.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/env python3 -u
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements functions to be used in evaluating classification models."""
+
+__author__ = ["ksharma6", "jgyasu"]
+__all__ = ["evaluate"]
+
+import time
+import warnings
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import KFold
+
+from sktime.datatypes import check_is_scitype, convert
+from sktime.exceptions import FitFailedWarning
+from sktime.split import InstanceSplitter
+from sktime.utils.dependencies import _check_soft_dependencies
+from sktime.utils.multiindex import apply_split
+from sktime.utils.parallel import parallelize
+from sktime.utils.validation.forecasting import check_scoring
+
+PANDAS_MTYPES = ["pd.DataFrame", "pd.Series", "pd-multiindex", "pd_multiindex_hier"]
+
+
+def _check_scores(metrics) -> dict:
+    """Validate and coerce to BaseMetric and segregate them based on predict type.
+
+    Parameters
+    ----------
+    metrics : sktime accepted metrics object or a list of them or None
+
+    Return
+    ------
+    metrics_type : Dict
+        The key is metric types and its value is a list of its corresponding metrics.
+    """
+    if not isinstance(metrics, list):
+        metrics = [metrics]
+
+    metrics_type = {}
+    for metric in metrics:
+        metric = check_scoring(metric)
+        # collect predict type
+        if hasattr(metric, "get_tag"):
+            scitype = metric.get_tag(
+                "scitype:y_pred", raise_error=False, tag_value_default="pred"
+            )
+        else:  # If no scitype exists then metric is a point forecast type
+            scitype = "pred"
+        if scitype not in metrics_type.keys():
+            metrics_type[scitype] = [metric]
+        else:
+            metrics_type[scitype].append(metric)
+    return metrics_type
+
+
+def _get_column_order_and_datatype(
+    metric_types: dict, return_data: bool = True, cutoff_dtype=None, old_naming=True
```

Comment:
this is copy-paste from forecasting - I think the default should be `old_naming=False`, and all logic related to `old_naming=True` should be removed.